tests/models/test_gemini_vertex.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations as _annotations
+
+import json
+from pathlib import Path
+
+import httpx
+import pytest
+
+from pydantic_ai import Agent
+from pydantic_ai.messages import AudioUrl, DocumentUrl, ImageUrl, VideoUrl
+from pydantic_ai.models.gemini import GeminiModel
+
+from ..conftest import ClientWithHandler, TestEnv, try_import
+
+with try_import() as imports_successful:
+    from pydantic_ai.providers.google_vertex import GoogleVertexProvider
+
+
+pytestmark = [
+    pytest.mark.skipif(not imports_successful(), reason='google-auth not installed'),
+    pytest.mark.anyio,
+]
+
+
+async def mock_refresh_token():
+    return 'my-token'
+
+
+@pytest.mark.parametrize(
+    'url',
+    [
+        pytest.param(AudioUrl(url='https://cdn.openai.com/API/docs/audio/alloy.wav'), id='AudioUrl'),
+        pytest.param(AudioUrl(url='gs://my-bucket/path/to/audio-blob.mp3'), id='AudioUrl (gs://)'),
+        pytest.param(
+            DocumentUrl(url='https://storage.googleapis.com/cloud-samples-data/generative-ai/pdf/2403.05530.pdf'),
+            id='DocumentUrl',
+        ),
+        pytest.param(DocumentUrl(url='gs://my-bucket/path/to/pdf-blob.pdf'), id='DocumentUrl (gs://)'),
+        pytest.param(
+            ImageUrl(url='https://upload.wikimedia.org/wikipedia/commons/6/6a/Www.wikipedia_screenshot_%282021%29.png'),
+            id='ImageUrl',
+        ),
+        pytest.param(ImageUrl(url='gs://my-bucket/path/to/image-blob.png'), id='ImageUrl (gs://)'),
```

Comment:
Awesome to see that this PR intends to support GCS URIs like `gs://cloud-samples-data/generative-ai/image/scones.jpg`!